test/python/primitives/test_result.py

```diff
@@ -0,0 +1,93 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for BasePrimitiveResult."""
+
+from __future__ import annotations
+
+from collections.abc import Collection
+from dataclasses import dataclass
+from typing import Any
+
+from ddt import data, ddt, unpack
+
+from qiskit.primitives.base_result import BasePrimitiveResult
+from qiskit.test import QiskitTestCase
+
+
+################################################################################
+## STUB DATACLASS
+################################################################################
+@dataclass
+class Result(BasePrimitiveResult):
+    """Dummy result dataclass implementing BasePrimitiveResult."""
+
+    field_1: Collection[Any]
+    field_2: Collection[Any]
+
+
+################################################################################
+## TESTS
+################################################################################
+@ddt
+class TestBasePrimitiveResult(QiskitTestCase):
+    """Tests BasePrimitiveResult."""
+
+    @data(0, 1.2, True, "sequence", {"name": "value"})
```

Comment:
Need to check `bytes` too?
```suggestion
    @data(0, 1.2, True, "sequence", b"sequence", {"name": "value"})
```